### PR TITLE
chore: accomodate iot custom auth tests

### PR DESCRIPTION
**Pull request by jamesonwilliams**
Adds facilities to the `aws-android-sdk-iot-test` suite to make use of
custom authorizers in test.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws-android-sdk-iot-test/src/androidTest/java/com/amazonaws/mobileconnectors/iot/CustomAuthorization.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+    String getTokenValue() {
+        return tokenValue;
+    }
+
+    String getTokenSignature() {
+        return tokenSignature;
+    }
+
+    public String getCustomAuthorizerName() {
```
Comment (jpignata) [inline aws-android-sdk-iot-test/src/androidTest/java/com/amazonaws/mobileconnectors/iot/CustomAuthorization.java:98]: Sorry, dumb question maybe, is this visibility difference intentional?

File: aws-android-sdk-iot-test/src/androidTest/java/com/amazonaws/mobileconnectors/iot/IotClient.java [hunk 2]
```diff
@@ -254,6 +234,48 @@ boolean deletePolicy(String policyName) {
...
+        request.setTokenSigningPublicKeys(Collections.singletonMap("FIRST_KEY", publicKey));
+        iot.createAuthorizer(request);
+    }
+
+    void cleanupAuthorizer(String authorizerName) {
+        try {
+            updateAuthorizer(authorizerName, AuthorizerStatus.INACTIVE);
+            deleteAuthorizer(authorizerName);
+        } catch (ResourceNotFoundException noSuchAuthorizer) {
+            // Oh well.
```
Comment (jpignata) [inline aws-android-sdk-iot-test/src/androidTest/java/com/amazonaws/mobileconnectors/iot/IotClient.java:262]: Win some, lose some.

File: aws-android-sdk-iot-test/src/androidTest/java/com/amazonaws/mobileconnectors/iot/MqttManagerIntegrationTest.java [hunk 3]
```diff
@@ -625,15 +641,15 @@ public void onMessageArrived(String topic, byte[] data) {
         mqttManager.publishString("please kill me", "$aws", AWSIotMqttQos.QOS0);
```
Comment (jpignata) [inline aws-android-sdk-iot-test/src/androidTest/java/com/amazonaws/mobileconnectors/iot/MqttManagerIntegrationTest.java:641]: Dag, yo.

File: aws-android-sdk-iot-test/src/androidTest/java/com/amazonaws/mobileconnectors/iot/CustomAuthorization.java (hunk 1 above)
Comment (jamesonwilliams) [reply aws-android-sdk-iot-test/src/androidTest/java/com/amazonaws/mobileconnectors/iot/CustomAuthorization.java:98]: Nope! Not intentional. I've just been sloppy with them today. :-D Will fix.

File: aws-android-sdk-iot-test/src/androidTest/java/com/amazonaws/mobileconnectors/iot/IotClient.java (hunk 2 above)
Comment (jamesonwilliams) [reply aws-android-sdk-iot-test/src/androidTest/java/com/amazonaws/mobileconnectors/iot/IotClient.java:262]: Ill do the `true`/`false` boolean return thing here, so the info is not lost.

File: aws-android-sdk-iot-test/src/androidTest/java/com/amazonaws/mobileconnectors/iot/MqttManagerIntegrationTest.java (hunk 3 above)
Comment (jamesonwilliams) [reply aws-android-sdk-iot-test/src/androidTest/java/com/amazonaws/mobileconnectors/iot/MqttManagerIntegrationTest.java:641]: haha I hadn't noticed this.

**Pull request merged**